tests/Integration/Subscriber/Optimization/DequeueJQueryMigrateSubscriber/dequeueJqueryMigrate.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace WP_Rocket\Tests\Integration\Subscriber\Optimization\Dequeue_JQuery_Migrate_Subscriber;
+
+use Brain\Monkey;
+use WP_Rocket\Tests\Integration\TestCase;
+
+/**
+ * @covers WP_Rocket\Subscriber\Optimization\Dequeue_JQuery_Migrate_Subscriber::dequeue_jquery_migrate
+ *
+ * @group  jQueryMigrate
+ */
+class Test_DequeueJqueryMigrate extends TestCase {
+	private $orig_deps;
+	private $wp_scripts;
+	private $count;
+
+	public function setUp() {
+		parent::setUp();
+
+		$wp_scripts       = wp_scripts();
+		$this->orig_deps  = $wp_scripts->registered['jquery']->deps;
+		$this->wp_scripts = $wp_scripts;
+		$this->count      = did_action( 'wp_default_scripts' );
+	}
+
+	public function tearDown() {
+		parent::tearDown();
+
+		// Restore.
+		wp_scripts()->registered['jquery']->deps = $this->orig_deps;
+	}
+
+	public function testShouldDequeueJqueryMigrate() {
+		add_filter( 'pre_get_rocket_option_dequeue_jquery_migrate', '__return_true' );
+
+		$this->wp_scripts->init();
```

Comment:
`wp_scripts()` instantiates once. 

```php
	function wp_scripts() {
		global $wp_scripts;
		if ( ! ( $wp_scripts instanceof WP_Scripts ) ) {
			$wp_scripts = new WP_Scripts(); // WPCS: override ok.
		}
		return $wp_scripts;
	}
```

We need to `init` each time to run the code and rebuild the scripts. Why?

Reply:
Indeed `wp_scripts()` cannot be used because we need to run this hook `wp_default_scripts`.
This hook is called only in `init` of `WP_Scripts`, so for each test we need to reinit `WP_Scripts`, to make sure that our subscriber is working properly and our callback is called.
Another solution could be to:
- run in a separate process each test for dequeue jquery migrate ( which will be time consuming)
- reload Wordpress using the helper `$this->go_to()` ( is time comsuming too )